pkg/cloudprovider/aws/fake/ec2api.go

```diff
@@ -91,13 +98,13 @@ func (e *EC2API) DescribeLaunchTemplatesWithContext(ctx context.Context, input *
 		return e.DescribeLaunchTemplatesOutput, nil
 	}
 	output := &ec2.DescribeLaunchTemplatesOutput{}
-	for _, wanted := range input.LaunchTemplateNames {
-		for _, launchTemplate := range e.LaunchTemplates {
-			if launchTemplate.LaunchTemplateName == wanted {
-				output.LaunchTemplates = append(output.LaunchTemplates, launchTemplate)
-			}
+	e.LaunchTemplates.Range(func(key, value interface{}) bool {
+		launchTemplate := value.(*ec2.LaunchTemplate)
+		if functional.ContainsString(aws.StringValueSlice(input.LaunchTemplateNames), aws.StringValue(launchTemplate.LaunchTemplateName)) {
+			output.LaunchTemplates = append(output.LaunchTemplates, launchTemplate)
 		}
-	}
+		return true
```

Comment:
Why would this function always be returning true? Wouldn't we want it to only return true if it does contain the string?

Reply:
The Range function continues if you return true. We want to process the entire list, so we always return true. The check for adding the LT to the list is on line 103.